lib/markdown_generators.js

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env node
+
+const yaml = require("js-yaml");
+const markdown = require("markdown-builder");
+const title_case = require("title-case");
+const TurndownService = require("turndown");
+const tables = require("turndown-plugin-gfm").tables;
+const turndown_service = new TurndownService();
+turndown_service.use(tables);
+const helpers = require("./helpers");
+
+class markdown_generators {
+    static generate_course_home_front_matter(course_data) {
+        /*
+        Generate the front matter metadata for the course home page given course_data JSON
+        */
+        let front_matter = {};
```

Comment:
We should assign the values in the initial object, to the extent possible:

    const front_matter = {
      title: "Course Home",
      course_title: course_data['title'],
      course_image_url: helpers.get_course_image_url(course_data),
      ...
    }